src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -327,10 +327,17 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
                 _innerBuilder.Services.AddSingleton<DashboardServiceHost>();
                 _innerBuilder.Services.AddHostedService(sp => sp.GetRequiredService<DashboardServiceHost>());
                 _innerBuilder.Services.AddSingleton<IDashboardEndpointProvider, HostDashboardEndpointProvider>();
-                _innerBuilder.Services.AddLifecycleHook<DashboardLifecycleHook>();
                 _innerBuilder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<DashboardOptions>, ConfigureDefaultDashboardOptions>());
                 _innerBuilder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IValidateOptions<DashboardOptions>, ValidateDashboardOptions>());
 
+                _innerBuilder.Services.AddSingleton<DashboardInjectionService>();
+                Eventing.Subscribe<BeforeStartEvent>((e, ct) =>
+                {
+                    var injectionService = e.Services.GetRequiredService<DashboardInjectionService>();
+                    injectionService.InjectDashboard(e.Model);
+                    return Task.CompletedTask;
+                });
+
```

Comment:
@davidfowl looking at this change I'm not entirely certain we should get rid of `IDistributedApplicationLifecycleHook` completely. There are instances where that approach works really well (such as these global things like injecting the dashboard).

I think where `TrySubscribeOnce` comes in handy is more for hosting integrations where you may have multiple of a resource.